app/src/main/res/layout-land/popular_movies_fragment.xml

```diff
@@ -29,22 +29,32 @@
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toTopOf="parent" />
 
-        <androidx.recyclerview.widget.RecyclerView
-            android:id="@+id/recyclerViewPopularMoviesFragment"
+        <androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+            android:id="@+id/popularMoviesRefreshOnSwipe"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:adapter="@{recyclerAdapterForPopularMovies}"
-            android:clipToPadding="false"
-            android:paddingBottom="88dp"
-            app:adapterData="@{viewModel.popularMovies}"
-            app:layoutManager="androidx.recyclerview.widget.GridLayoutManager"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
-            app:spanCount="2"
-            tools:listitem="@layout/popular_movie_recycler_item_layout"
-            tools:visibility="visible" />
+            android:layout_height="wrap_content"
+            app:refreshing="@{viewModel.isStartingRefresh}"
+            app:onRefreshListener="@{() -> viewModel.refreshPopularMovie()}"
+            tools:ignore="MissingConstraints">
```

Comment:
What is the purpose of this ignore?

Reply:
No, I don't have a purpose for it, I added constraints